src/roster-history.js

```diff
@@ -0,0 +1,57 @@
+class RosterHistory {
+
+	static get() {
+		let historyStore = localStorage.getItem('roster-history');
+		historyStore = !historyStore ? { history: [] } : JSON.parse(historyStore);
+		return historyStore;
+	}
+
+	static showList() {
+		const historyEl = document.getElementById('history');
+		const historyListEl = document.getElementById('history-list');
+
+		let rosterHistory = this.get();
+		let html = '';
+		if (rosterHistory.history.length) {
+			rosterHistory.history.forEach((roster, index) => {
+				html += ` 
+					<div class='roster-history-box'>
+						<div class='backdrop'">
+							<div class="load-wrapper" onclick="Main.rerun(${index})"><b>LOAD</b></div>
+							<div class='delete' onclick="RosterHistory.remove(${index})"><b>X</b></div>
+						</div>
+						<span><b>Name:</b> ${roster.name}</span>
+						<span><b>Battlesize:</b> ${roster.battleSize}</span>
+						<span><b>Faction:</b> ${roster.faction.name}</span>
+						<span><b>Poins:</b> ${roster.faction.points}</span>
+					</div>`;
+			});
+			historyListEl.innerHTML = html;
+			historyEl.removeAttribute('hidden');
+		} else {
+			historyEl.setAttribute('hidden', true);
+		}
+	}
+
+	static add(roster) {
+		const historyStore = this.get();
+		historyStore.history.unshift(roster);
```

Comment:
Should we check if the roster (or a roster with same name, size, faction and points) already exists before adding a new one?

Reply:
I wouldnt do it right now... probably need additional confirmation for that as well.. since technically if you're didnt name the roster it can overwrite some different one... then we need to compare whole rosters together...

Reply:
But then you might end up with 10 1k rosters all named `New Roster` and no idea which is which 😁

Reply:
there is a delete button :)